exchange/bitswap/bitswap.go

```diff
@@ -339,6 +339,24 @@ func (bs *bitswap) ReceiveMessage(ctx context.Context, p peer.ID, incoming bsmsg
 	return "", nil
 }
 
+// Connected/Disconnected warns bitswap about peer connections
+func (bs *bitswap) PeerConnected(p peer.ID) {
+	// TODO: add to clientWorker??
+
+	peers := make(chan peer.ID)
+	err := bs.sendWantlistToPeers(context.TODO(), peers)
```

Comment:
This will hang. It reads on the channel, which you dont fill until after the call